src/libYARP_init/src/CustomInit.cpp

```diff
@@ -37,18 +38,32 @@ extern "C" void yarpCustomFini()
     yarp::os::LogForwarder::clearInstance();
 }
 
+
 yarp::os::Network::Network() {
     Network::init();
 }
 
+yarp::os::Network::Network(yarp::os::yarpClockType clockType, yarp::os::Clock *custom) {
+    init(clockType, custom);
+}
+
 yarp::os::Network::~Network() {
     Network::fini();
 }
 
 void yarp::os::Network::init() {
+    init(yarp::os::YARP_CLOCK_DEFAULT);
+}
+
+
+void yarp::os::Network::init(yarp::os::yarpClockType clockType, yarp::os::Clock *custom) {
     if (__custom_yarp_is_initialized==0) {
         initMinimum();
+        // If we init the clock inside the initMinum, it will loop into itself twice
+        // calling again the initMinimun, ending with __yarp_is_initialized counter 
+        // increased twice.
         yarpCustomInit();
+        Network::yarpClockInit(clockType, custom);
```

Comment:
This is not good, this MUST be called inside `initMinimum()` or we well end in situations where the clock is not initialized.
Can you explain me why should initMinimum loop into itself?